uat/custom-components/client-python-paho/src/grpc_client_server/grpc_control_server.py

```diff
@@ -115,13 +131,118 @@
         context - request context
         Returns MqttConnectReply object
         """
-        # TODO
-        self.__logger.info("CreateMqttConnection Placeholder TODO")
+        client_id = request.clientId
+        if (client_id is None) or (len(client_id) == 0):

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`

[Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)

Similar issue at line numbers 140, 192, 196, and 200.

Reply:
fixed in c1b9703